src/isec_surf_surf.py

```diff
@@ -1,4 +1,284 @@
+                    sp_i = u2_index * surf2.v_basis.n_intervals + v2_index
+                    # v2 fixed
+                    u2i = np.array([surf2.u_basis.knots[u2_index+2], surf2.u_basis.knots[u2_index+3]])
+                    v2_thread = np.linspace(surf2.v_basis.knots[v2_index+2],surf2.v_basis.knots[v2_index+3],nt)
+                    for v2_val  in v2_thread:  #=1:length(v_knot)
+                        u2_center = (u2i[0] + u2i[1])/2
+                        v2i = np.array([v2_val])
+                        uvu2v2 = np.array([u1_center, v1_center, u2_center])#initial condition
+                        uvu2v2,ptl,conv  = self._patch_patch_intersection( surf1, surf2,uvu2v2, ui,vi,u2i,v2i, nit ,u2_index,v2_index,u_index,v_index)
+
+                        if np.not_equal(conv, 0):
+                            s = s+1
+                            n_points = n_points +1
+                            point[n_points,:] = [uvu2v2,v2_val,ptl(1),ptl(2),ptl(3),u2_index,v2_index,u_index,v_index]
+
+                        # u2 fixed
+                        #v2i = [vs_knots(o+2) vs_knots(o+3)]
+                        #u_knot = linspace(us_knots(m+2),us_knots(m+3),nt)
+                        #for h =1:length(u_knot)
+                        #    v2_c = (vs_knots(o+2) + vs_knots(o+3))/2
+                        #    u2i = u_knot(h)
+                        #    uvu2v2 = [u1_c,v1_c,v2_c]; % initial condition
+                        #    [ uvu2v2,ptl,conv ] = patch_patch_intersection( uvu2v2, ui,vi,u2i,v2i, u_knots, v_knots, X,us_knots, vs_knots, Xs, nit ,m,o,k,l)
+                        #    if conv ~= 0
+                        #        s = s+1
+                        #        n_points = n_points +1
+                        #        point(n_points,:) = [uvu2v2(1:2)',u_knot(h),uvu2v2(3),ptl(1),ptl(2),ptl(3),k,l,m,o]
+
+                ninter[u_index,v_index] = s
+
+        return point, n_points, ninter
+
+    def _patch_patch_intersection( self,surf1, surf2,uvt, ui,vi,u2i,v2i, nit ,m,o,k,l):
+        #returns coordinetes of the intersection of the patches (if exist), one
+        #parameter must be fixed
+        pt = np.zeros([3,1])
+        conv =0
+        tol = 1e-6 # in x,y,z
+        tol2 = 1e-4 # in u,v
+        # nt = self.nt
+        #print(uvt.shape)
+        #print(uvt[0])
+        #print(uvt)
+        #print(m,o,k,l)
+
+        X2 = surf2.poles[m:m + 3, o:o + 3,:]
+        X = surf1.poles[k:k + 3, l:l + 3,:]
+
+
+        if u2i.shape[0] == 1:
+            u2f = surf2.u_basis.eval_vector(m,u2i) # eval_vector(self, i_base, t):
+
+        if v2i.shape[0] == 1:
+            v2f = surf2.v_basis.eval_vector(o,v2i)
+
+        for i in range(nit):
+            uf = surf1.u_basis.eval_vector(k,uvt[0])
+            vf = surf1.v_basis.eval_vector(l,uvt[1])
+            ufd = surf1.u_basis.eval_diff_vector(k,uvt[0])
+            vfd = surf1.v_basis.eval_diff_vector(l,uvt[1])
+
+            if u2i.shape[0] == 1:
```

Comment:
Chapat to jako prunik plochy a krivky a pak je jedno zda je to hlavni krivka v U nebo V. Vidlicky se zrusi a nahrani predpocitanim polu krivky pred volanim metody.